src/Model/ValueObject/Identity/UserId.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace CultuurNet\UDB3\Model\ValueObject\Identity;
+
+use CultuurNet\UDB3\Model\ValueObject\String\Behaviour\IsNotEmpty;
+use CultuurNet\UDB3\Model\ValueObject\String\Behaviour\IsString;
+
+final class UserId
+{
+    use IsString;
+    use IsNotEmpty;
+
+    public function __construct(string $value)
+    {
+        $this->guardNotEmpty($value);
```

Comment:
Would it be a good idea to add some regex here to check if it has a valid structure of an id?

Reply:
I think that could become a difficult regex because we have several formats to support, for example:
- `google-oauth2|102486314601596809843`
- `auth0|813ff497-f0b8-4a59-bd44-dc2646ad98ff`
- `auth0|631748dba64ea78e3983b207`
- `9e68dafc-01d8-4c1c-9612-599c918b981d`

But I could create some basic if statements to check prefix and/or uuid?

Reply:
If we add validation I would opt for if checks for the different prefixes. To support these different patterns a regex would be to fragile.

Reply:
maybe check either if it is in a list of allowed prefixes
and then if nothing else if it is a regex?

Reply:
The problem is that the list of allowed prefixes is not static, we have `auth0`, `google-oauth2`, `facebook`, `apple`, `csam` in the past we also had `linkedin`. In the future, new services might be added, so just a valid string seems the logical approach to me.

Reply:
Good to know this! But I guess this can be something we expose somewhere while doing the migration to Keycloack?